### PR TITLE
feat: improve tampering test on exercise "Authentication System"

### PR DESCRIPTION
**Pull request by salvatore-piccione**
Tests for `Authentication System` exercises 4 and 5 can be improved by trying to actually tamper the data to ensure that the expected changes have been done. This PR adds two test cases:

- the former is about the tampering check on the admin
- the latter is about the tampering of developers by replacing an element in the dictionary

Exemplar solution has been updated to be more inline with the exercise request (i.e. `getDevelopers()` returns an instance of the interface `IDictionary` despite of an instance of `ReadOnlyDictionary`).

### CODE REVIEW

**Comment by salvatore-piccione** [comment]
Hello @ErikSchierboom, I saw that checks on Configlet failed. AFAIK this is not due by the changes on this PR since the failure affects other exercises.

File: exercises/concept/authentication-system/AuthenticationSystemTests.cs [hunk 1]
```diff
@@ -14,6 +14,17 @@ public void GetAdmin()
...
 
+    [Fact]
+    [Task(4)]
+    public void CheckAdminCannotBeTampered()
+    {
+        var admin = new Identity { EyeColor = "green", Email = "admin@ex.ism" };
+        var authenticator = new Authenticator(admin);
+        var tamperedAdmin = authenticator.Admin;
+        tamperedAdmin.Email = "admin@hack.ed";
+        Assert.NotEqual(tamperedAdmin.Email, authenticator.Admin.Email);
```
Comment (ErikSchierboom) [inline exercises/concept/authentication-system/AuthenticationSystemTests.cs:25]: Maybe do:

```suggestion
        Assert.Equal("admin@ex.ism", authenticator.Admin.Email);
```

This is a slightly stronger assertion, as the current test theoretically still passes when the admin email is changed to something else than the tampered admin.

File: exercises/concept/authentication-system/AuthenticationSystemTests.cs [hunk 2]
```diff
@@ -24,4 +35,22 @@ public void GetDevelopers()
...
+
+        Identity tamperedDev = new Identity { EyeColor = "grey", Email = "anders@hack.ed" };
+        try
+        {
+            devs["Anders"] = tamperedDev;
+            Assert.True(false, "Unexpected change to developers list.");
+        } catch (NotSupportedException)
+        {
+            Assert.True(true);            
+        }
```
Comment (ErikSchierboom) [inline exercises/concept/authentication-system/AuthenticationSystemTests.cs:54]: This should also work I think:

```suggestion
        Assert.Throws<NotSupportedException>(() => devs["Anders"] = tamperedDev);
```

File: exercises/concept/authentication-system/AuthenticationSystemTests.cs (hunk 1 above)
Comment (salvatore-piccione) [reply exercises/concept/authentication-system/AuthenticationSystemTests.cs:25]: Totally agree!

File: exercises/concept/authentication-system/AuthenticationSystemTests.cs (hunk 2 above)
Comment (salvatore-piccione) [reply exercises/concept/authentication-system/AuthenticationSystemTests.cs]: Changes done.

File: exercises/concept/authentication-system/AuthenticationSystemTests.cs (hunk 1 above)
Comment (salvatore-piccione) [reply exercises/concept/authentication-system/AuthenticationSystemTests.cs]: Changes done.

**Review by ErikSchierboom — APPROVED** [review]
Thanks!

**Pull request merged**